mono/profiler/proflog.c

```diff
@@ -3103,7 +3100,6 @@ typedef struct MonoCounterAgent {
 } MonoCounterAgent;
 
 static MonoCounterAgent* counters;
-static gboolean counters_initialized = FALSE;
```

Comment:
Because in the same commit, I moved helper thread initialization to just after counters initialization, and made sure only the helper thread samples counters. So counters will always be initialized when we sample them.